images/calico/kind-tests/main.tf

```diff
@@ -0,0 +1,99 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+    kind = {
+      source  = "tehcyx/kind"
+      version = ">=0.0.16"
+    }
+    kubernetes = {}
+    helm       = {}
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = ">=1.14.0"
+    }
+  }
+}
+
+data "http" "file" {
+  for_each = toset([
+    "https://raw.githubusercontent.com/projectcalico/calico/v3.26.1/manifests/calico-typha.yaml",
+    "https://raw.githubusercontent.com/projectcalico/calico/v3.26.1/manifests/csi-driver.yaml"
+  ])
+
+  url = each.key
+}
+
+data "kubectl_file_documents" "docs" {
+  for_each = data.http.file
+
+  content = each.value.response_body
+}
+
+locals {
+  yaml_file = flatten([
+    for doc in data.kubectl_file_documents.docs : [
+      for v in doc.documents : {
+        data : yamldecode(v)
+        content : v
+      }
+    ]
+  ])
+}
+
+provider "kubernetes" {
+  host                   = kind_cluster.no_cni.endpoint
+  client_certificate     = kind_cluster.no_cni.client_certificate
+  client_key             = kind_cluster.no_cni.client_key
+  cluster_ca_certificate = kind_cluster.no_cni.cluster_ca_certificate
+}
+
+provider "kubectl" {
+  host                   = kind_cluster.no_cni.endpoint
+  client_certificate     = kind_cluster.no_cni.client_certificate
+  client_key             = kind_cluster.no_cni.client_key
+  cluster_ca_certificate = kind_cluster.no_cni.cluster_ca_certificate
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = kind_cluster.no_cni.endpoint
+    client_certificate     = kind_cluster.no_cni.client_certificate
+    client_key             = kind_cluster.no_cni.client_key
+    cluster_ca_certificate = kind_cluster.no_cni.cluster_ca_certificate
+  }
+}
+
+resource "kubectl_manifest" "install" {
+  for_each = {
+    for v in local.yaml_file : lower(join("/", compact([
+      v.data.apiVersion,
+      v.data.kind,
+      lookup(lookup(v.data, "metadata", {}), "namespace", ""),
+      lookup(lookup(v.data, "metadata", {}), "name", "")
+    ]))) => v.content
+  }
+  yaml_body = each.value
+
+  depends_on = [kind_cluster.no_cni]
+}
+
+resource "kind_cluster" "no_cni" {
+  name = "kind-no-cni"
+
+  # This needs to be disabled, the provider checks assume a default CNI
+  # wait_for_ready = true
+
+  kind_config {
+    kind        = "Cluster"
+    api_version = "kind.x-k8s.io/v1alpha4"
+    networking {
+      disable_default_cni = true
+    }
+  }
+}
+
+data "oci_exec_test" "smoke" {
```

Comment:
Let's make a `var` for this image, with the default being `:latest`, and pass it through an `oci_ref` to resolve it to digest before passing here.

Then we can have a simple repro of `:latest` with the opportunity to override it with whatever other image we might be testing.

Reply:
great idea! pushed with latest changes that actually use our images 😅